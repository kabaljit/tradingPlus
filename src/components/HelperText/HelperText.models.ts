export interface IHelperTextProps {
  type?: 'info' | 'error';
  visible?: boolean;
  testID?: string;
}
