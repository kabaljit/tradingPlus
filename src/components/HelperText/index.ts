export { HelperText as default } from './HelperText';
