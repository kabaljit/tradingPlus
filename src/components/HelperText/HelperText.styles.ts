import styled from 'styled-components/native';

import theme from '../../theme';
import {
  defaultTextInputHelperPaddingVertical,
  defaultTextInputHelperPaddingHorizontal,
  defaultTextInputHelperFontSize,
} from '../TextInput';

import { IHelperTextProps } from './HelperText.models';

const getColor = (props: IHelperTextProps) => {
  const visible = props.visible === true || props.visible === undefined;
  if (!visible) return 'transparent';
  return props.type === 'error'
    ? theme.context.customColors.input.border.error
    : theme.context.customColors.input.border.default
};

export const HelperTextView = styled.View`
  height: ${defaultTextInputHelperFontSize +
  defaultTextInputHelperPaddingVertical * 2}px;
`;

// TODO: Find a better name. Possibly the component itself shouldn't be HelperText to avoid confusion.
export const HelperTextText = styled.Text<IHelperTextProps>`
  padding-vertical: ${defaultTextInputHelperPaddingVertical}px;
  padding-horizontal: ${defaultTextInputHelperPaddingHorizontal}px;
  font-size: ${defaultTextInputHelperFontSize}px;
  color: ${getColor};
`;
