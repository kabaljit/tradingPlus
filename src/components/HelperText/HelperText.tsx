import * as React from 'react';

import { IHelperTextProps } from './HelperText.models';
import { HelperTextView, HelperTextText } from './HelperText.styles';

export const HelperText: React.FunctionComponent<IHelperTextProps> = (
  props
) => {
  const { children, ...propsRest } = props;
  return (
    <HelperTextView>
      <HelperTextText {...propsRest}>{children}</HelperTextText>
    </HelperTextView>
  );
};
