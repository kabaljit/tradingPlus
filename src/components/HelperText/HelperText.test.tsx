import React from 'react';
import renderer from 'react-test-renderer';

import { HelperText } from './HelperText';

describe('Given a HelperText', () => {
  describe('when the component is rendered', () => {
    it('should match the snapshot with default values', () => {
      const tree = renderer.create(<HelperText>HELP</HelperText>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match the snapshot when not visible', () => {
      const tree = renderer
        .create(<HelperText visible={false}>HELP</HelperText>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match the snapshot when displaying an error', () => {
      const tree = renderer
        .create(<HelperText type="error">HELP!!!!</HelperText>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
