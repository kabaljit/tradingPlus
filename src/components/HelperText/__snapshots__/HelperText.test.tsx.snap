// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a HelperText when the component is rendered should match the snapshot when displaying an error 1`] = `
<View
  style={
    Array [
      Object {
        "height": 27.900000000000002,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFE0D7",
          "fontSize": 17.271428571428572,
          "paddingHorizontal": 5.314285714285715,
          "paddingVertical": 5.314285714285715,
        },
      ]
    }
    type="error"
  >
    HELP!!!!
  </Text>
</View>
`;

exports[`Given a HelperText when the component is rendered should match the snapshot when not visible 1`] = `
<View
  style={
    Array [
      Object {
        "height": 27.900000000000002,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "transparent",
          "fontSize": 17.271428571428572,
          "paddingHorizontal": 5.314285714285715,
          "paddingVertical": 5.314285714285715,
        },
      ]
    }
    visible={false}
  >
    HELP
  </Text>
</View>
`;

exports[`Given a HelperText when the component is rendered should match the snapshot with default values 1`] = `
<View
  style={
    Array [
      Object {
        "height": 27.900000000000002,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#9699AA",
          "fontSize": 17.271428571428572,
          "paddingHorizontal": 5.314285714285715,
          "paddingVertical": 5.314285714285715,
        },
      ]
    }
  >
    HELP
  </Text>
</View>
`;
